src/deploy/functions/cache/hash.ts

```diff
@@ -2,52 +2,37 @@ import { readFile } from "node:fs/promises";
 import * as crypto from "crypto";
 import { Backend, Endpoint } from "../backend";
 import { getSecretVersions } from "../../../functions/secrets";
+import { BinaryLike } from "crypto";
 
 /**
  * Generates a hash from the environment variables of a {@link Backend}.
  * @param backend Backend of a set of functions
  */
 export function getEnvironmentVariablesHash(backend: Backend): string {
-  const hash = crypto.createHash("sha256");
-
   // Hash the contents of the dotenv variables
   const hasEnvironmentVariables = !!Object.keys(backend.environmentVariables).length;
-  if (hasEnvironmentVariables) {
-    hash.update(JSON.stringify(backend.environmentVariables));
-  }
-
-  return hash.digest("hex");
+  return createHash(hasEnvironmentVariables ? JSON.stringify(backend.environmentVariables) : "");
```

Comment:
nit* hasEnviromentVariable feels slightly less useful w/ the refactored code. Seems like it's equilvalent to write something like

```
return createHash(JSON.stringify(backend.environmentVariables || {}))
```

(though this might break your tests. I think it's cool to ignore this comment if this feels too picky)

Reply:
You're right, `JSON.stringify({})` isn't any more or less useful than `""`.  Updated! (And tests didn't break!)